synapse/app/synctl.py

```diff
@@ -190,7 +199,11 @@ def main():
         if start_stop_synapse:
             stop(pidfile, "synapse.app.homeserver")
 
-        # TODO: Wait for synapse to actually shutdown before starting it again
+    # Wait for synapse to actually shutdown before starting it again
+    if action == "restart":
+        write("Waiting to restart ...")
```

Comment:
a log line is useful, it just needs to say something different.

"Waiting for process to exit before restarting...", for example